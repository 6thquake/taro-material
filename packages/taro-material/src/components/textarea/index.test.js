import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'

import AtTextarea from '../../../.temp/components/textarea/index'

describe('AtTextarea Snap', () => {
  it('render initial AtTextarea', () => {
    const componet = renderToString(<AtTextarea isTest />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTextarea -- props customStyle', () => {
    const componet = renderToString(<AtTextarea isTest customStyle='color:red;' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTextarea -- props className', () => {
    const componet = renderToString(<AtTextarea isTest className='test' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTextarea -- props value', () => {
    const componet = renderToString(<AtTextarea isTest value='test' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTextarea -- props maxlength', () => {
    const componet = renderToString(<AtTextarea isTest maxlength='300' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTextarea -- props maxlength', () => {
    const componet = renderToString(<AtTextarea isTest placeholder='test' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTextarea -- props maxlength', () => {
    const componet = renderToString(<AtTextarea isTest count={3000} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTextarea -- props maxlength', () => {
    const componet = renderToString(<AtTextarea isTest height={3000} />)
    expect(componet).toMatchSnapshot()
  })
})
