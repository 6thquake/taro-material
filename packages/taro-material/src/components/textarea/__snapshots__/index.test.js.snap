// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtTextarea Snap render AtTextarea -- props className 1`] = `"<div class=\\"at-textarea test\\"><textarea placeholderClass=\\"placeholder\\" cursorSpacing=\\"100\\" value=\\"\\" confirmType=\\"完成\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" class=\\"at-textarea__textarea\\" placeholder=\\"\\" maxlength=\\"200\\"></textarea><div class=\\"at-textarea__bottom\\">0/200</div></div>"`;

exports[`AtTextarea Snap render AtTextarea -- props customStyle 1`] = `"<div style=\\"color:red;\\" class=\\"at-textarea\\"><textarea placeholderClass=\\"placeholder\\" cursorSpacing=\\"100\\" value=\\"\\" confirmType=\\"完成\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" class=\\"at-textarea__textarea\\" placeholder=\\"\\" maxlength=\\"200\\"></textarea><div class=\\"at-textarea__bottom\\">0/200</div></div>"`;

exports[`AtTextarea Snap render AtTextarea -- props maxlength 1`] = `"<div class=\\"at-textarea\\"><textarea placeholderClass=\\"placeholder\\" cursorSpacing=\\"100\\" value=\\"\\" confirmType=\\"完成\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" class=\\"at-textarea__textarea\\" placeholder=\\"\\" maxlength=\\"300\\"></textarea><div class=\\"at-textarea__bottom\\">0/300</div></div>"`;

exports[`AtTextarea Snap render AtTextarea -- props maxlength 2`] = `"<div class=\\"at-textarea\\"><textarea placeholderClass=\\"placeholder\\" cursorSpacing=\\"100\\" value=\\"\\" confirmType=\\"完成\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" class=\\"at-textarea__textarea\\" placeholder=\\"test\\" maxlength=\\"200\\"></textarea><div class=\\"at-textarea__bottom\\">0/200</div></div>"`;

exports[`AtTextarea Snap render AtTextarea -- props maxlength 3`] = `"<div class=\\"at-textarea\\"><textarea placeholderClass=\\"placeholder\\" cursorSpacing=\\"100\\" value=\\"\\" confirmType=\\"完成\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" class=\\"at-textarea__textarea\\" placeholder=\\"\\" maxlength=\\"200\\"></textarea><div class=\\"at-textarea__bottom\\">0/200</div></div>"`;

exports[`AtTextarea Snap render AtTextarea -- props maxlength 4`] = `"<div class=\\"at-textarea\\"><textarea style=\\"height:64rem\\" placeholderClass=\\"placeholder\\" cursorSpacing=\\"100\\" value=\\"\\" confirmType=\\"完成\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" class=\\"at-textarea__textarea\\" placeholder=\\"\\" maxlength=\\"200\\"></textarea><div class=\\"at-textarea__bottom\\">0/200</div></div>"`;

exports[`AtTextarea Snap render AtTextarea -- props value 1`] = `"<div class=\\"at-textarea\\"><textarea placeholderClass=\\"placeholder\\" cursorSpacing=\\"100\\" value=\\"test\\" confirmType=\\"完成\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" class=\\"at-textarea__textarea\\" placeholder=\\"\\" maxlength=\\"200\\"></textarea><div class=\\"at-textarea__bottom\\">4/200</div></div>"`;

exports[`AtTextarea Snap render initial AtTextarea 1`] = `"<div class=\\"at-textarea\\"><textarea placeholderClass=\\"placeholder\\" cursorSpacing=\\"100\\" value=\\"\\" confirmType=\\"完成\\" selectionStart=\\"-1\\" selectionEnd=\\"-1\\" class=\\"at-textarea__textarea\\" placeholder=\\"\\" maxlength=\\"200\\"></textarea><div class=\\"at-textarea__bottom\\">0/200</div></div>"`;
